### PR TITLE
[Misc] Replace api to get feature size in `node_classfication` and `hetero_rgcn`.

### PR DESCRIPTION
**Pull request by yxy235**
## Description

Replace API to get feature size.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/sampling/graphbolt/lightning/node_classification.py [hunk 1]
```diff
@@ -209,8 +208,15 @@ def val_dataloader(self):
...
-    datamodule = DataModule([15, 10, 5], args.batch_size, args.num_workers)
-    model = SAGE(100, 256, datamodule.num_classes).to(torch.double)
+    dataset = gb.BuiltinDataset("ogbn-products").load()
+    datamodule = DataModule(
+        [15, 10, 5],
+        args.batch_size,
+        args.num_workers,
+        dataset,
+    )
+    in_feat = dataset.feature.size("node", None, "feat")[0]
```
Comment (frozenbugs) [inline examples/sampling/graphbolt/lightning/node_classification.py:218]: in_feat -> in_size

**Review by frozenbugs — APPROVED** [review]

File: examples/sampling/graphbolt/lightning/node_classification.py [hunk 2]
```diff
@@ -133,12 +133,11 @@ def configure_optimizers(self):
 
 
 class DataModule(LightningDataModule):
-    def __init__(self, fanouts, batch_size, num_workers):
+    def __init__(self, fanouts, batch_size, num_workers, dataset):
```
Comment (frozenbugs) [inline examples/sampling/graphbolt/lightning/node_classification.py:136]:
```suggestion
    def __init__(self, dataset, fanouts, batch_size, num_workers):
```

File: examples/sampling/graphbolt/lightning/node_classification.py [hunk 3]
```diff
@@ -209,8 +208,15 @@ def val_dataloader(self):
...
 
-    datamodule = DataModule([15, 10, 5], args.batch_size, args.num_workers)
-    model = SAGE(100, 256, datamodule.num_classes).to(torch.double)
+    dataset = gb.BuiltinDataset("ogbn-products").load()
+    datamodule = DataModule(
+        [15, 10, 5],
+        args.batch_size,
+        args.num_workers,
+        dataset,
+    )
```
Comment (frozenbugs) [inline examples/sampling/graphbolt/lightning/node_classification.py:217]:
```suggestion
    datamodule = DataModule(
        dataset,
        [15, 10, 5],
        args.batch_size,
        args.num_workers,
    )
```

File: examples/sampling/graphbolt/lightning/node_classification.py [hunk 4]
```diff
@@ -133,7 +133,7 @@ def configure_optimizers(self):
 
 
 class DataModule(LightningDataModule):
-    def __init__(self, fanouts, batch_size, num_workers, dataset):
+    def __init__(self, dataset, fanouts, batch_size, num_workers):
```
Comment (frozenbugs) [inline examples/sampling/graphbolt/lightning/node_classification.py:136]: @yxy235 primary parameters should come first in the parameter list. FYI.

Comment (yxy235) [reply examples/sampling/graphbolt/lightning/node_classification.py:136]: Get.

> @yxy235 primary parameters should come first in the parameter list. FYI.

**Pull request merged**